File: languages/csharp/AppEncryption/AppEncryption.Tests/AppEncryption/AppEncryptionBytesImplTest.cs
using System;
using GoDaddy.Asherah.AppEncryption.Envelope;
using Moq;
using Xunit;

namespace GoDaddy.Asherah.AppEncryption.Tests.AppEncryption
{
    [Collection("Logger Fixture collection")]
    public class AppEncryptionBytesImplTest
    {
        private readonly Mock<IEnvelopeEncryption<string>> envelopeEncryptionMock;
        private readonly AppEncryptionBytesImpl<string> appEncryptionBytesImpl;

        public AppEncryptionBytesImplTest()
        {
            envelopeEncryptionMock = new Mock<IEnvelopeEncryption<string>>();
            appEncryptionBytesImpl = new AppEncryptionBytesImpl<string>(envelopeEncryptionMock.Object);
        }

        [Fact]
        private void TestConstructor()
        {
            AppEncryptionBytesImpl<string> appEncryption = new AppEncryptionBytesImpl<string>(envelopeEncryptionMock.Object);
            Assert.NotNull(appEncryption);
        }

        [Fact]
        private void TestDecrypt()
        {
            byte[] expectedBytes = { 0, 1, 2, 3, 4 };

            envelopeEncryptionMock.Setup(x => x.DecryptDataRowRecord(It.IsAny<string>())).Returns(expectedBytes);

            byte[] actualBytes = appEncryptionBytesImpl.Decrypt("some data row record");
            Assert.Equal(expectedBytes,  actualBytes);
        }

        [Fact]
        private void TestEncrypt()
        {
            const string expectedDataRowRecord = "some data row record";

            envelopeEncryptionMock.Setup(x => x.EncryptPayload(It.IsAny<byte[]>())).Returns(expectedDataRowRecord);

            string actualDataRowRecord = appEncryptionBytesImpl.Encrypt(new byte[] { 0, 1, 2, 3, 4 });
            Assert.Equal(expectedDataRowRecord, actualDataRowRecord);
        }

        [Fact]
        private void TestDispose()
        {
            appEncryptionBytesImpl.Dispose();
            envelopeEncryptionMock.Verify(x => x.Dispose());
        }

        [Fact]
        private void TestCloseWithCloseFailShouldReturn()
        {
            envelopeEncryptionMock.Setup(x => x.Dispose()).Throws<SystemException>();
            appEncryptionBytesImpl.Dispose();
            envelopeEncryptionMock.Verify(x => x.Dispose());
        }
    }
}

File: languages/java/app-encryption/src/test/java/com/godaddy/asherah/appencryption/AppEncryptionBytesImplTest.java
package com.godaddy.asherah.appencryption;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import com.godaddy.asherah.appencryption.envelope.EnvelopeEncryption;

@ExtendWith(MockitoExtension.class)
class AppEncryptionBytesImplTest {

  @Mock
  EnvelopeEncryption<String> envelopeEncryption;
  @InjectMocks
  AppEncryptionBytesImpl<String> appEncryptionBytesImpl;
  
  @Test
  void testConstructor() {
    AppEncryption<?, ?> appEncryption = new AppEncryptionBytesImpl<>(envelopeEncryption);
    assertNotNull(appEncryption);
  }

  @Test
  void testDecrypt() {
    byte[] expectedBytes = new byte[]{0, 1, 2, 3};
    when(envelopeEncryption.decryptDataRowRecord(any())).thenReturn(expectedBytes);

    byte[] actualBytes = appEncryptionBytesImpl.decrypt("some data row record");
    assertArrayEquals(expectedBytes, actualBytes);
  }

  @Test
  void testEncrypt() {
    String expectedDataRowRecord = "some data row record";
    when(envelopeEncryption.encryptPayload(any())).thenReturn(expectedDataRowRecord);

    String actualDataRowRecord = appEncryptionBytesImpl.encrypt(new byte[]{0, 1, 2, 3, 4});
    assertEquals(expectedDataRowRecord, actualDataRowRecord);
  }

  @Test
  void testCloseSuccess() {
    appEncryptionBytesImpl.close();
    
    // Verify proper resources are closed
    verify(envelopeEncryption).close();
  }

  @Test
  void testCloseWithCloseFailShouldReturn() {
    doThrow(RuntimeException.class).when(envelopeEncryption).close();
    appEncryptionBytesImpl.close();
    
    verify(envelopeEncryption).close();
  }

}
